docs/illink-options.md

```diff
@@ -0,0 +1,184 @@
+# IL Linker Command Line Options
+
+## illink tool
+
+The illink is IL Linker version shipping with .NET Core or .NET 5.0 platforms. It's bundled with
+the .NET SDK and most of the options are accessible using msbuild properties but any option
+can also be passed using `ExtraArgs` property.
+
+### Linking from the main assembly
+
+The command:
+
+`illink -a Program.exe`
+
+will use the assembly Program.exe as a source. That means that the linker will
+walk through all the methods of Program.exe to generate only what is necessary
+for this assembly to run.
+
+### Linking from an [XML descriptor](data-formats.md#descriptor-format)
+
+The command:
+
+`illink -x desc.xml`
+
+will use the XML descriptor as a source. That means that the linker will
+use this file to decide what to link in a set of assemblies. The format of the
+descriptors is described further on in this document.
```

Comment:
```suggestion
descriptors is described in [data-formats document](../data-formats.md).
```

Please make sure that the link actually works - I don't know if relative paths like this are supported by GitHub.